src/java.base/share/classes/java/lang/reflect/Executable.java

```diff
@@ -691,17 +692,34 @@ AnnotatedType getAnnotatedReturnType0(Type returnType) {
      * @jls 8.8 Constructor Declarations
      */
     public AnnotatedType getAnnotatedReceiverType() {
-        if (Modifier.isStatic(this.getModifiers()))
+        if (Modifier.isStatic(this.getModifiers())) {
             return null;
+        }
+        Class<?> c = getDeclaringClass();
+        TypeVariable<?>[] v = c.getTypeParameters();
```

Comment:
Same here, can this be pushed down to a common method that either returns a Class or resolves the appropriate Type instance?

Reply:
I moved it to an annotation factory class and avoided the import.